src/main/java/frc/robot/subsystems/MoverSubsystem.java

```diff
@@ -0,0 +1,130 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMax.SoftLimitDirection;
+
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.motorcontrol.MotorUtil;
+import static frc.robot.Constants.MoverConstants.*;
+
+public class MoverSubsystem extends SubsystemBase{
+
+    private final CANSparkMax rotationMotor;
+    private final RelativeEncoder rotationEncoder;
+    private final SparkMaxPIDController rotationPidController;
+
+    private final CANSparkMax extensionMotor;
+    private final RelativeEncoder extensionEncoder;
+    private final SparkMaxPIDController extensionPidController;
+
+    private final double ROTATION_P = 0.125;
+    private final double ROTATION_I = 0;
+    private final double ROTATION_D = 0;
+    private final double ROTATION_ROT_TO_RAD = ROTATION_GEAR_RATIO * 2 * Math.PI;
+
+    private final double EXTENSION_P = 0.125;
+    private final double EXTENSION_I = 0;
+    private final double EXTENSION_D = 0;
+    private final double EXTENSION_ROT_TO_M = Units.inchesToMeters(0.5615); 
+
+    private final boolean RESET_OFFSET_ON_STAGE_SWITCH = true;
+    private double angleOffset = 0; //radians
+    private double extensionOffset = 0; //meters
+
+    private MoverPosition currentState = MoverPosition.GROUND;
+
+    public enum GamePiece {
+        CONE, CUBE;
+    }
+    
+    public enum MoverPosition {
+        VERTICAL(0,0),
+        GROUND(Units.degreesToRadians(90), distanceToExtension(25)),
+        SUBSTATION(0,0),
+        CUBEMID(0,0), CUBEHIGH(0,0), 
+        CONEMID(0,0), CONEHIGH(0,0)
+        ;
+
+        public double angle; //Radians
+        public double extension; //Meters, 0 extension is 25 inches from pivot
+
+        private MoverPosition(double angle, double extension){
+            this.angle = angle;
+            this.extension = extension;
+        }
+    }
+
+    public MoverSubsystem(){
+        rotationMotor = MotorUtil.createSparkMax(ROTATION_MOTOR_PORT);
+        rotationMotor.setIdleMode(IdleMode.kBrake);
+
+        rotationEncoder = rotationMotor.getEncoder();
+        rotationEncoder.setPositionConversionFactor(ROTATION_ROT_TO_RAD);
+        
+        rotationMotor.enableSoftLimit(SoftLimitDirection.kForward, true);
+        rotationMotor.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        rotationMotor.setSoftLimit(SoftLimitDirection.kForward, (float) Units.degreesToRadians(90));
```

Comment:
won't it take more than 90 degrees to get to the ground? i'm guessing 0 degrees is straight up

Reply:
I think it will depend on the grabber, we can definitely change that later. I just didn't want to accidently slam into the ground on the first test.